### PR TITLE
Update libp2p::protocol::common::Scheduler with libp2p::basic::Scheduler

### PR DESCRIPTION
**Pull request by Alexey-N-Chernyshov**
Signed-off-by: Alexey-N-Chernyshov <alexey.n.chernyshov@gmail.com>

### Description of the Change
Updates libp2p::protocol::common::Scheduler with libp2p::basic::Scheduler.

### CODE REVIEW

File: core/miner/storage_fsm/impl/sealing_impl.cpp [hunk 1]
```diff
@@ -132,23 +131,21 @@ namespace fc::mining {
...
+          sealing->scheduler_->schedule(
+              [self{sealing}, sector_id = sector.second->sector_number]() {
+                auto maybe_error = self->startPacking(sector_id);
+                if (maybe_error.has_error()) {
+                  self->logger_->error("starting sector {}: {}",
+                                       sector_id,
+                                       maybe_error.error().message());
+                }
+              },
+              config.wait_deals_delay);
```
Comment (Alexey-N-Chernyshov) [inline core/miner/storage_fsm/impl/sealing_impl.cpp:148]: There was a bug with constant `config.max_wait_deals_sectors`.

**Review by turuslan — APPROVED** [review]

File: core/miner/storage_fsm/impl/sealing_impl.cpp [hunk 2]
```diff
@@ -326,40 +323,37 @@ namespace fc::mining {
...
-        .detach();
+      std::vector<Piece> pieces;
+      for (auto &piece : maybe_pieces.value()) {
+        pieces.push_back(Piece{
+            .piece = std::move(piece),
+            .deal_info = boost::none,
+        });
+      }
+
+      auto maybe_error = self->newSectorWithPieces(sid, pieces);
```
Comment (wer1st) [inline core/miner/storage_fsm/impl/sealing_impl.cpp:352]:
```suggestion
      const auto maybe_error = self->newSectorWithPieces(sid, pieces);
```

Comment (wer1st) [reply core/miner/storage_fsm/impl/sealing_impl.cpp:352]: here and below

File: core/storage/ipfs/graphsync/impl/network/length_delimited_message_reader.cpp [hunk 3]
```diff
@@ -49,10 +49,10 @@ namespace fc::storage::ipfs::graphsync {
     libp2p::basic::VarintReader::readVarint(
         stream_,
         [wptr{weak_from_this()}]
-            (boost::optional<libp2p::multi::UVarint> varint_opt) {
+            (outcome::result<libp2p::multi::UVarint> varint_opt) {
           auto self = wptr.lock();
           if (self && self->reading_) {
-            size_t length = varint_opt ? varint_opt->toUInt64() : 0;
+            size_t length = varint_opt ? varint_opt.value().toUInt64() : 0;
```
Comment (wer1st) [inline core/storage/ipfs/graphsync/impl/network/length_delimited_message_reader.cpp:55]:
```suggestion
            const size_t length = varint_opt ? varint_opt.value().toUInt64() : 0;
```

**Review by wer1st — APPROVED** [review]

File: core/miner/storage_fsm/impl/sealing_impl.cpp (hunk 2 above)
Comment (Alexey-N-Chernyshov) [reply core/miner/storage_fsm/impl/sealing_impl.cpp]: done
